atomic_reactor/plugins/pre_reactor_config.py

```diff
@@ -0,0 +1,363 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import unicode_literals
+
+from atomic_reactor.plugin import PreBuildPlugin
+
+from collections import namedtuple
+import logging
+import yaml
+
+
+# Key used to store the config object in the plugin workspace
+WORKSPACE_CONF_KEY = 'reactor_config'
+
+
+def get_config(workflow):
+    """
+    Obtain configuration object
+    Does not fail
+
+    :return: ReactorConfig instance
+    """
+    try:
+        workspace = workflow.plugin_workspace[ReactorConfigPlugin.key]
+        return workspace[WORKSPACE_CONF_KEY]
+    except KeyError:
+        # The plugin did not run or was not successful: use defaults
+        return ReactorConfig()
+
+
+class ConfigBuilder(object):
```

Comment:
Usually that's what Json and YAML are for. If the user creates a config that uses the incorrect type, and validation fails, I think that should be treated as an error. 

IMO, if the config format used allows different types, we should leverage, and enforce, that. This is different, for instance, than INI files via python's ConfigParser which pretty much treats all values as strings.